src/main/java/org/jenkinsci/test/acceptance/junit/JenkinsAcceptanceTestRule.java

```diff
@@ -53,14 +53,28 @@ public void evaluate() throws Throwable {
 
                 injector.injectMembers(this);
 
+                Throwable throwable = null;
                 try {
                     decorateWithRules(base).evaluate();
                 } catch (AssumptionViolatedException e) {
                     System.out.printf("Skipping %s%n", description.getDisplayName());
                     e.printStackTrace();
-                    throw e;
+                    throwable = e;
+                } catch (Throwable t) {
+                    throwable = t;
                 } finally {
-                    world.endTestScope();
+                    try {
+                        world.endTestScope();
+                    } catch (Throwable t) {
+                        if (throwable == null) {
+                            throwable = t;
+                        } else {
+                            throwable.addSuppressed(t);
```

Comment:
Using `.addSuppresed` here to ensure that the visual display of the error focuses on the testcase error rather than the cleanup error.